### PR TITLE
docs: update helm ref docs and connect docs

### PR DESCRIPTION
**Pull request by ishustava**
All k8s connect-related docs now need to mention that we require a Kubernetes service
for all Connect services

### CODE REVIEW

File: website/content/docs/k8s/connect/health.mdx [hunk 1]
```diff
@@ -9,6 +9,9 @@ description: Configuring Kubernetes Health Checks
 -> 0.26+: This feature is available in consul-helm versions 0.26 and higher and is defaulted on.
 To disable it, set `connectInject.healthChecks.enabled: false`.
 
+-> **Note**: As of consul-k8s `v0.26.0-beta1` and Consul Helm `v0.32.0-beta1`, syncing of the Kubernetes readiness status
```
Comment (kschoche) [inline website/content/docs/k8s/connect/health.mdx:12]: Nice. 
I think for GA we should clean up this whole page.

**Review by kschoche — APPROVED** [review]
🤿 
Love it, great work!

File: website/content/docs/k8s/connect/health.mdx (hunk 1 above)
Comment (ishustava) [reply website/content/docs/k8s/connect/health.mdx:12]: Agree!

File: website/content/docs/k8s/connect/index.mdx [hunk 2]
```diff
@@ -93,40 +123,67 @@ This is useful to transition to Connect by allowing both Connect and
...
 ### Connecting to Connect-Enabled Services
 
-The example pod specification below configures a pod that is capable
+The example Deployment specification below configures a Deployment that is capable
 of establishing connections to our previous example "static-server" service. The
 connection to this static text service happens over an authorized and encrypted
 connection via Connect.
 
+-> **Note:** As of consul-k8s `v0.26.0-beta1` and Consul Helm `v0.32.0-beta1`, having a Kubernetes
+service is **required** to run services on the Consul Service Mesh.
```
Comment (lkysow) [inline website/content/docs/k8s/connect/index.mdx:142]:
```suggestion
Service is **required** to run services on the Consul Service Mesh.
```

**Review by lkysow — APPROVED** [review]

**Pull request merged**